src/Microsoft.Framework.Runtime.Compilation.DesignTime/DesignTimeHostCompiler.cs

```diff
@@ -26,6 +26,7 @@ public DesignTimeHostCompiler(IApplicationShutdown shutdown, IFileWatcher watche
                 foreach (var file in files)
                 {
                     watcher.WatchFile(file);
+                    watcher.WatchProject(file);
```

Comment:
The internals of WatchProject already check if the project is being watched and if not it grabs the project path from file and adds a watcher for the project.